### PR TITLE
noderestriction: add alpha config group

### PR DESCRIPTION
**Pull request by mikedanese**

### CODE REVIEW

**Comment by tallclair** [comment]
This doesn't feel like it justifies a whole new API group. Can this just be added to the componentconfig group? Alternatively, maybe this should be a CRD?

What's the bootstrapping flow?

**Comment by luxas** [comment]
cc @kubernetes/sig-cluster-lifecycle-pr-reviews @kubernetes/sig-auth-pr-reviews

**Comment by ericchiang** [comment]
> This doesn't feel like it justifies a whole new API group. Can this just be added to the componentconfig group? Alternatively, maybe this should be a CRD?

@tallclair I think this is intended to be a config schema like the audit policy API, not a full API served by the API server. Could have sworn we discussed this in https://github.com/kubernetes/community/pull/911 but I'm having trouble finding that comment.

**Comment by liggitt** [comment]
The more 911 progressed, the less it seemed we needed these levers. We should wait until we need them to introduce this.

**Comment by mikedanese** [comment]
Sorry for the stall. @liggitt and I were discussing whether we needed the API group at all. @liggitt suggested [here](https://github.com/kubernetes/community/pull/911#discussion_r176927423) to use an admission controller configuration.

File: plugin/pkg/admission/noderestriction/apis/noderestriction/types.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Configuration provides configuration for the NodeRestriction admission
+// controller.
+type Configuration struct {
```
Comment (php-coder) [inline plugin/pkg/admission/noderestriction/apis/noderestriction/types.go]: This name looks too generic. Consider renaming it to be more specific.

Comment (mikedanese) [reply plugin/pkg/admission/noderestriction/apis/noderestriction/types.go]: It would be redundant with the API group name.

**Pull request closed without merging**